### PR TITLE
Feature/3537 PayawayCreate message signing

### PR DESCRIPTION
**Pull request by DavidRamsbottomRTGS**

### CODE REVIEW

File: src/RTGS.DotNetSDK/RTGS.DotNetSDK.IntegrationTests/Publisher/Signing/GivenOpenConnection/WhenSigningIsSuccessful.cs [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+		{
+			{ "pairwise-did-signature", SignDocument.Response.PairwiseDidSignature },
+			{ "public-did-signature", SignDocument.Response.PublicDidSignature },
+			{ "alias", TestData.ValidMessages.IdCryptAlias }
+		};
+
+		receivedMessage.Headers.Should().BeEquivalentTo(expectedHeaders);
+	}
+
+	private record SignDocumentRequest<TDocument>
```
Comment (josephreillyltd) [inline src/RTGS.DotNetSDK/RTGS.DotNetSDK.IntegrationTests/Publisher/Signing/GivenOpenConnection/WhenSigningIsSuccessful.cs:185]: Why is this a private record in this class?

Comment (DavidRamsbottomRTGS) [reply src/RTGS.DotNetSDK/RTGS.DotNetSDK.IntegrationTests/Publisher/Signing/GivenOpenConnection/WhenSigningIsSuccessful.cs:185]: It models the body of the IDCrypt API call.  The real model is internal to the ID Crypt SDK.  We have been making private copies in the test classes to help with testing.

File: src/RTGS.DotNetSDK/RTGS.DotNetSDK.IntegrationTests/Publisher/Signing/GivenOpenConnection/WhenSigningIsSuccessful.cs [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+	public async Task WhenCallingIdCryptAgent_ThenConnectionIdIsInBody<TRequest>(PublisherAction<TRequest> publisherAction)
+	{
+		_toRtgsMessageHandler.SetupForMessage(handler => handler.ReturnExpectedAcknowledgementWithSuccess());
+
+		await publisherAction.InvokeSendDelegateAsync(_rtgsPublisher);
+
+		var requestContent = await _idCryptMessageHandler.Requests[SignDocument.Path]
+			.Single().Content.ReadAsStringAsync();
+
+		var signDocumentRequest = JsonSerializer.Deserialize<SignDocumentRequest<TRequest>>(requestContent, new JsonSerializerOptions { PropertyNameCaseInsensitive = true });
```
Comment (RobBentleyRtgs) [inline src/RTGS.DotNetSDK/RTGS.DotNetSDK.IntegrationTests/Publisher/Signing/GivenOpenConnection/WhenSigningIsSuccessful.cs:142]: long line

File: src/RTGS.DotNetSDK/RTGS.DotNetSDK/Publisher/IdCrypt/Signing/SignPayawayCreateMessage.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
+﻿using RTGS.IDCryptSDK.JsonSignatures;
+using RTGS.IDCryptSDK.JsonSignatures.Models;
+using RTGS.ISO20022.Messages.Pacs_008_001.V10;
+
+namespace RTGS.DotNetSDK.Publisher.IdCrypt.Signing;
+
+internal class SignPayawayCreateMessage : ISignMessage<FIToFICustomerCreditTransferV10>
```
Comment (josephreillyltd) [inline src/RTGS.DotNetSDK/RTGS.DotNetSDK/Publisher/IdCrypt/Signing/SignPayawayCreateMessage.cs:7]: I think we should rename this

File: src/RTGS.DotNetSDK/RTGS.DotNetSDK/Publisher/InternalPublisher.cs [hunk 4]
```diff
@@ -42,7 +47,11 @@ public InternalPublisher(ILogger<InternalPublisher> logger, Payment.PaymentClien
 
 		ArgumentNullException.ThrowIfNull(message, nameof(message));
 
+
```
Comment (josephreillyltd) [inline src/RTGS.DotNetSDK/RTGS.DotNetSDK/Publisher/InternalPublisher.cs:50]: extra line

File: src/RTGS.DotNetSDK/RTGS.DotNetSDK.IntegrationTests/Publisher/Signing/GivenOpenConnection/WhenSigningIsNotSuccessful.cs [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+		_serilogContext.PublisherLogs(LogEventLevel.Information)
+			.Should().ContainSingle().Which.Should().BeEquivalentTo(
+				new LogEntry($"Signing {typeof(TRequest).Name} message", LogEventLevel.Information));
+
+		_serilogContext.PublisherLogs(LogEventLevel.Error)
+				.Should().ContainSingle().Which.Should().BeEquivalentTo(
+					new LogEntry($"Error signing {typeof(TRequest).Name} message", LogEventLevel.Error, typeof(RtgsPublisherException)));
+	}
+
+	private record SignDocumentRequest<TDocument>
```
[the remark this answers is not in the document]
Comment (DavidRamsbottomRTGS) [reply src/RTGS.DotNetSDK/RTGS.DotNetSDK.IntegrationTests/Publisher/Signing/GivenOpenConnection/WhenSigningIsNotSuccessful.cs:138]: Done

File: src/RTGS.DotNetSDK/RTGS.DotNetSDK.IntegrationTests/Publisher/Signing/GivenOpenConnection/WhenSigningIsSuccessful.cs (hunk 2 above)
Comment (DavidRamsbottomRTGS) [reply src/RTGS.DotNetSDK/RTGS.DotNetSDK.IntegrationTests/Publisher/Signing/GivenOpenConnection/WhenSigningIsSuccessful.cs:142]: Done

File: src/RTGS.DotNetSDK/RTGS.DotNetSDK/Publisher/IdCrypt/Signing/SignPayawayCreateMessage.cs (hunk 3 above)
Comment (DavidRamsbottomRTGS) [reply src/RTGS.DotNetSDK/RTGS.DotNetSDK/Publisher/IdCrypt/Signing/SignPayawayCreateMessage.cs:7]: Done

File: src/RTGS.DotNetSDK/RTGS.DotNetSDK/Publisher/InternalPublisher.cs (hunk 4 above)
Comment (DavidRamsbottomRTGS) [reply src/RTGS.DotNetSDK/RTGS.DotNetSDK/Publisher/InternalPublisher.cs:50]: Done

**Review by josephreillyltd — APPROVED** [review]

**Pull request merged**